pages/glossary/changes-in-content.md

```diff
@@ -0,0 +1,28 @@
+---
+title: Changes in content
+key: changes-in-content
+unambiguous: true
+objective: true
+input_aspects:
+  - Accessibility tree
+  - Audio output
+  - CSS styling
+  - DOM tree
+---
+
+A _change in the content_ of a [web page][] caused by an event occurs when, by comparing the [web page][] before and after the event firing, at least one of the following occurs:
+
+- **visible changes:** the rendered pixels change in any part of the document that is currently within the [viewport][] or that can be brought into the [viewport][] via scrolling; or
+- **accessibility tree changes:** any state, property or event of a node representing an [accessible object][] of the [accessibility tree][] changes, or any node is inserted in, or removed from the [accessibility tree][]; or
+- **audible changes:** the audio rendered by the [web page][] changes.
+
+**Assumptions:**
+
+- If the [web page][] is rendering time-based media, rendered pixels and audio will be changing as part of the playback. The comparison in this instance should compare the pixels and audio that are rendered if the event is not fired, with the ones that are rendered if the event is fired.
+- For the purpose of this definition, it is assumed that the event causing the changes is the only origin of changes to the [web page][] and therefore the page does not change if there is no such event.
```

Comment:
That assumption seems very strong in its current state.
If two different events create changes, the assumption means that they can't happen (almost) simultaneously.

If I come back to the scenario you describe in https://github.com/act-rules/act-rules.github.io/pull/1276#discussion_r440710851 then once the time based media start, anything else (such as tilting to go to next song) will fail this assumption (there is another origin of changes). This is more or less caught by she previous point (ignoring changes that would have occurred anyway).

We both seem uneasy with this assumption (and/or the combination with the time based media note above). So we probably need to do something about it… I am not sure that completely removing it is the way to go…

---
In this definition, "event" can be almost anything (not just a DOM event). So I agree that changes that happen will be caused by some event, possibly far away (e.g. playing a 10h video will cause changes for 10h).

We need a way to figure out that a given change is caused by a given event. This is currently the note about the time based media, but could apply to other changes (e.g. caused by an asynchronous request completing, this can potentially create change long after the event that caused it, unless we consider that the request completion is the "event" (and not sending the request)).

This does illustrate that there is some degree of subjectivity (hopefully not ambiguity) in the definition. "event" not being a DOM event creates place for interpretation, hence subjectivity. This is not a problem but the frontmatter should be updated accordingly.

---
I thing we're pretty solid on what a "change" is (i.e., objective, unambiguous). But not so on what an "event" is (if not DOM event) and consequently on the causality. Maybe trying to get a better definition of "event" will help us figure out problems with this assumption?

Reply:
I hope that you agree that fixing this to dom events solves the problem :-)